### ISSUE TITLE
Problematic handling of Tables read from an external Database whose column names are aliased due to Unicode normalization

### ISSUE THREAD

**Issue by radeusgd** [enso-org/enso#7415]
### The problem

This is essentially an extremely rare edge case, but we need to handle it somehow - the moment it happens in production everyone will be super confused.

The issue is - most tools do not equalize strings modulo Unicode normalization (e.g. `'s\u0301' == 'ś'`) like Enso does. This means, that one can have a table with two columns **separate**: `ś` and `s\u0301` and once Enso loads a table like this, it will not be able to work with it. This should be very rare in practice, and indicates that the database we are working with is a **huge mess**. But it can technically happen.

To illustrate, I create such messy table ~~using raw SQL expressions~~ apparently I can even use our `Connection.create_table` as it does not have any special handling for this edge case.

```
from Standard.Base import all
from Standard.Table import all
from Standard.Database import all

main =
    c = Database.connect (SQLite In_Memory)

    s1 = 's\u0301'
    s2 = 'ś'

    IO.println (s1 == s2) # True
    IO.println (s1.codepoints != s2.codepoints) # True

    IO.println <|
        #c.execute_update 'CREATE TEMPORARY TABLE test123 ("'+s1+'" INTEGER, "'+s2+'" INTEGER)'
        c.create_table "test123" [Column_Description.Value s1 Value_Type.Integer, Column_Description.Value s2 Value_Type.Integer] temporary=True
    # (Database Table test123)
    IO.println <|
        c.execute_update 'INSERT INTO test123 VALUES (10, 20)'
    # 1

    t = c.query "test123"
    IO.println t # (Database Table test123)
    names = t.column_names
    IO.println names # [ś, ś]
    # Enso cannot distinguish columns of this table from SQLite
    IO.println ((names.at 0) == (names.at 1)) # True

    # And thus it is unable to materialize it
    IO.println t.print # (Error: (Illegal_Argument.Error 'Column names must be distinct.' Nothing))
    IO.println t.read  # (Error: (Illegal_Argument.Error 'Column names must be distinct.' Nothing))

    # But it can read individual columns
    IO.println (t.columns.map .to_vector) # [[10], [20]]

    # Selecting one column selects both due to equality :)
    IO.println (t.select_columns s1 . columns . length) # 2

    # And at only allows us to ever access the first column, we can never see the other one :/
    IO.println (t.at s1 . to_vector) # [10]
    IO.println (t.at s2 . to_vector) # [10]

    # Apparently, set only affects first column!
    t2 = t.set "123" s1
    IO.println (t2.columns.map .to_vector) # [[123], [20]]
    # But regardless if we use s1 or s2, we can only manipulate the first column :/
    IO.println (t.set "456" s2 . columns . map .to_vector) # [[456], [20]]

```

As we can see, accessing such messy table does not fail but it proceeds to behave pretty badly.

1. Trying to materialize it, will construct an in-memory Table. But in-memory both columns have the same name so it will fail.
2. Even worse, operations on the table will behave in a surprising manner - e.g. `at` can ever reach the first column, because from Enso perspective they are indisinguishable.

#### Workaround

Working with such a messy table is currently very problematic - it can lead to surprising errors - like selecting wrong column (e.g. the dropdown for `at` will show both instances of the column, but selecting either of them will lead to getting the same column due to the `at` issue shown above), materializing won't work.

I think it is fair to detect this early - at the stage of `query` and report that this table is 'messy' and Enso cannot properly work with it. IMO that is better than returning a Table object that will partially work and can potentially lead to computing corrupted data (by using the wrong column etc.).

IMO reporting an error should be fine in this case, because it is really a mess - no one should have tables that have the same name just differing with one being Unicode normalized and the other not being normalized.

#### A 'defensive' solution

Still, ideally we want Enso to work correctly with whatever messy data is thrown at it. To achieve that, we can have special logic when querying tables that checks for such an edge case. It can then use our 'standard' renaming strategy to make the column names unique - e.g. `ś` and `ś 1`. Then on Enso side the column names would be distinguishable and Enso can deal normally with the data. Under the hood, we'd still have to keep an SQL expression for each column and be sure that the identifiers do not 'get normalized' by accident, to ensure that each of the two columns is mapped to the right column.

In case we implement that solution, we should emit a warning when accessing such messy tables, which would explain the situation and why one of the columns got renamed.

**Comment by radeusgd** [comment]
The simple workaround is essentially 1-2 lines of code to implement, so IMO we should get it in soon(ish) to avoid unnecessary confusion.

The more complicated solution may take more time, and given that this is such an extremely unlikely scenario, I don't think it's worth tackling very soon.

**Comment by radeusgd** [comment]
This was already mostly solved as part of #7495.

https://github.com/enso-org/enso/blob/abdbcfcdb10f28679d693232263aaed8826ff894/test/Table_Tests/src/Database/Postgres_Spec.enso#L238-L250

We are already reporting `Duplicate_Output_Column_Names` warning. It does not include a thorough explanation of the Unicode clash, but I think it provides the user with enough info - column names in the original table were clashing and so were enabled.
